lib/WorseReflection/Core/Cache/TtlCache.php

```diff
@@ -32,12 +32,21 @@ public function __construct(float $lifetime = 5.0)
     public function getOrSet(string $key, Closure $setter)
     {
         $now = microtime(true);
+        if (null === $this->lifetimeStart) {
+            $this->lifetimeStart = $now;
+        }
 
         if (isset($this->cache[$key]) && $this->expires[$key] > $now) {
             return $this->cache[$key];
         }
 
-        $this->purgeExpired($now);
+        $elapsed = $now - $this->lifetimeStart;
+
+        if ($elapsed >= $this->lifetime) {
```

Comment:
this coudl extend some cache entries to double the max lifetime, might be worth reducing this or doing it every X fetches